Content.Shared/Nutrition/EntitySystems/SatiationSystem.cs

```diff
@@ -0,0 +1,396 @@
+using System.Diagnostics.CodeAnalysis;
+using Content.Shared.Alert;
+using Content.Shared.Damage;
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Movement.Systems;
+using Content.Shared.Movement.Components;
+using Content.Shared.Nutrition.Components;
+using Content.Shared.Rejuvenate;
+using Content.Shared.StatusIcon;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Nutrition.EntitySystems;
+
+public sealed class SatiationSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly AlertsSystem _alerts = default!;
+    [Dependency] private readonly DamageableSystem _damageable = default!;
+    [Dependency] private readonly MobStateSystem _mobState = default!;
+    [Dependency] private readonly MovementSpeedModifierSystem _movementSpeedModifier = default!;
+    [Dependency] private readonly SharedJetpackSystem _jetpack = default!;
+
+    private (string, StatusIconPrototype?)[] HungerIcons = new (string, StatusIconPrototype?)[] {
+        ("HungerIconOverfed", null),
+        ("HungerIconPeckish", null),
+        ("HungerIconStarving", null)
+    };
+    /// <summary>
+    /// A dictionary relating hunger thresholds to corresponding alerts.
+    /// </summary>
+    private Dictionary<SatiationThreashold, AlertType> HungerAlertThresholds = new()
+    {
+        { SatiationThreashold.Concerned, AlertType.Peckish },
+        { SatiationThreashold.Desperate, AlertType.Starving },
+        { SatiationThreashold.Dead, AlertType.Starving }
+    };
+    private AlertCategory HungerAlertCategory = AlertCategory.Hunger;
+
+    private (string, StatusIconPrototype?)[] ThirstIcons = new (string, StatusIconPrototype?)[] {
+        ("ThirstIconOverhydrated", null),
+        ("ThirstIconThirsty", null),
+        ("ThirstIconParched", null)
+    };
+    /// <summary>
+    /// A dictionary relating hunger thresholds to corresponding alerts.
+    /// </summary>
+    private Dictionary<SatiationThreashold, AlertType> ThirstAlertThresholds = new()
+    {
+        { SatiationThreashold.Concerned, AlertType.Thirsty },
+        { SatiationThreashold.Desperate, AlertType.Parched },
+        { SatiationThreashold.Dead, AlertType.Parched }
+    };
+    private AlertCategory ThirstAlertCategory = AlertCategory.Thirst;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        foreach (var pair in ThirstIcons)
+        {
+            var (iconId, prototype) = pair;
+            DebugTools.Assert(_prototype.TryIndex(iconId, out prototype));
+        }
+        foreach (var pair in HungerIcons)
+        {
+            var (iconId, prototype) = pair;
+            DebugTools.Assert(_prototype.TryIndex(iconId, out prototype));
+        }
```

Comment:
when you merge master this wont be necessary anymore (alerts now properly use prototype ids)

Reply:
removed, now ids are directly passed to alert system